extensions/amp-font/0.1/amp-font.js

```diff
@@ -0,0 +1,327 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/**
+ * @fileoverview Triggers and monitors loading of custom fonts on AMP pages.
+ * Example:
+ * <code>
+ * <amp-font
+ *   font-family="My Font"
+ *   timeout="3s"
+ *   font-weight="bold"
+ *   on-error-add-class="myfont2-missing"
+ *   on-error-remove-class="myfont3-loaded"
+ *   on-load-add-class="myfont2-loaded"
+ *   on-load-remove-class="myfont1-loaded"
+ *   layout="nodisplay">
+ * </amp-font>
+ * </code>
+ *
+ * the amp-font element's layout type is nodisplay.
+ */
+
+import {removeElement} from '../../../src/dom';
+import * as style from '../../../src/style';
+import {timer} from '../../../src/timer';
+import {vsync} from '../../../src/vsync';
+
+/** @private @const {Array.<string>} */
+const DEFAULT_FONTS_ = ['sans-serif', 'serif'];
+
+/** @private @const {string} */
+const TEST_STRING_ = 'MAxmTYklsjo190QW';
+
+/** @private @const {number} */
+const TOLERANCE_ = 2;
+
+/** @private @const {number} */
+const DEFAULT_TIMEOUT_ = 5000;
+
+/** @private @const {string} */
+const DEFAULT_WEIGHT_ = '400';
+
+/** @private @const {string} */
+const DEFAULT_VARIANT_ = 'normal';
+
+/** @private @const {string} */
+const DEFAULT_STYLE_ = 'normal';
+
+/** @private @const {string} */
+const DEFAULT_SIZE_ = 'medium';
+
+
+export class AmpFont extends AMP.BaseElement {
+
+
+  /** @override */
+  prerenderAllowed() {
+    return true;
+  }
+
+
+  /** @override */
+  buildCallback() {
+    this.customFontFamily_ = this.element.getAttribute('font-family');
+    this.customFontWeight_ =
+        this.element.getAttribute('font-weight') || DEFAULT_WEIGHT_;
+    this.customFontStyle_ =
+        this.element.getAttribute('font-style') || DEFAULT_STYLE_;
+    this.customFontVariant_ =
+        this.element.getAttribute('font-variant') || DEFAULT_VARIANT_;
+    this.timeoutHandle_ = null;
+    this.hasTimedOut_ = false;
+    this.domElements_ = {
+      CONTAINER: null,
+      DEFAULT_FONT_ELEMENTS: [],
+      CUSTOM_FONT_ELEMENT: null
+    };
+    this.win_ = this.getWin();
```

Comment:
Do we need these vars? Why not just do calls where we need them or use local vars for them? This would apply to `this.win_`, `this.document_` and `this.documentElement_`.